presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -317,6 +317,17 @@ public int getRunningQueries()
         }
     }
 
+    private int getAggregatedRunningQueries()
+    {
+        int aggregatedRunningQueries = runningQueries.size() + descendantRunningQueries;
+        Optional<ResourceGroupRuntimeInfo> resourceGroupRuntimeInfo = getAdditionalRuntimeInfo();
+        if (resourceGroupRuntimeInfo.isPresent()) {
```

Comment:
Should we add a comment here that this if block is for disagg coordinator only ?

Do we still need any synchronization here ?

Also, can we add unit tests for this change ?

Reply:
- I don't think such comments are helpful, we have many place such branching logic.
- yes synchronization needed, going to add that with next iteraction, PR is out just to make sure no tests are failing. It's not ready for review.
- Yes working on adding unit tests. We don't have any unit tests around scheduling weight so need to add a new one.

Reply:
Apologies ! I did not realize that the PR is still WIP. This must have been a hard bug to track down :-)

Reply:
no worries, it's ready for review now.